packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/optimize-ciemss-operation.ts

```diff
@@ -166,42 +168,75 @@ export async function getOptimizedInterventions(optimizeRunId: string) {
 
-	if (interventionType === InterventionTypes.paramValue && startTime.length !== 0) {
-		// intervention type == parameter value
-		console.log(policyResult);
+	const allInterventions: Intervention[] = simulationIntervetions;
+
+	// TODO: https://github.com/DARPA-ASKEM/terarium/issues/3909
+	// This will need to be updated to allow multiple intervention types. This is not allowed at the moment.
+	if (interventionType === InterventionTypes.paramValue && startTimes.length !== 0) {
+		// If we our intervention type is param value our policyResult will provide a timestep.
 		for (let i = 0; i < paramNames.length; i++) {
-			// This is all index matching for optimizeInterventions.paramNames, optimizeInterventions.startTimes, and policyResult
-			// TODO: We will need to fix this for the interventions refactor
-			// simulationIntervetions.push({
-			// 	name: paramNames[i],
-			// 	timestep: startTime[i],
-			// 	value: policyResult[i]
-			// });
+			allInterventions.push({
+				name: `Optimized ${paramNames[i]}`,
+				appliedTo: paramNames[i],
+				type: InterventionSemanticType.Parameter,
+				staticInterventions: [
+					{
+						timestep: policyResult[i],
+						value: paramValues[i]
+					}
+				],
+				dynamicInterventions: []
+			});
 		}
-	} else if (interventionType === InterventionTypes.startTime && paramValue.length !== 0) {
+	} else if (interventionType === InterventionTypes.startTime && paramValues.length !== 0) {
+		// If we our intervention type is start time our policyResult will provide a value.
 		for (let i = 0; i < paramNames.length; i++) {
-			// This is all index matching for optimizeInterventions.paramNames, optimizeInterventions.startTimes, and policyResult
-			// TODO: We will need to fix this for the interventions refactor
-			// simulationIntervetions.push({
-			// 	name: paramNames[i],
-			// 	timestep: policyResult[i],
-			// 	value: paramValue[i]
-			// });
+			allInterventions.push({
+				name: `Optimized ${paramNames[i]}`,
+				appliedTo: paramNames[i],
+				type: InterventionSemanticType.Parameter,
+				staticInterventions: [
+					{
+						timestep: startTimes[i],
+						value: policyResult[i]
+					}
+				],
+				dynamicInterventions: []
+			});
 		}
 	} else {
 		// Should realistically not be hit unless we change the interface and do not update
 		console.error(`Unable to find the intevention for optimization run: ${optimizeRunId}`);
 	}
+	return allInterventions;
+}
 
-	return simulationIntervetions;
+export async function createInterventionPolicyFromOptimize(
+	modelConfigId: string,
+	optimizeRunId: string
+) {
+	const modelId = await getModelIdFromModelConfigurationId(modelConfigId);
+	const optimizedInterventions = await getOptimizedInterventions(optimizeRunId);
+
+	const newIntervention: InterventionPolicy = {
+		modelId,
+		interventions: optimizedInterventions
+	};
+	const newInterventionResponse: InterventionPolicy =
+		await createInterventionPolicy(newIntervention);
+	return newInterventionResponse;
```

Comment:
We are not dealing with a "Response" object per se, would be better to just call it "newInterventionPolicy" or something.

Reply:
Tangent, you are pulling in a good amount of service-level logic into the this operator file, be careful not to create circular dependencies.